src/transformState.js

```diff
@@ -5,7 +5,31 @@
  * @param {Object[]} actions
  */
 function transformState(state, actions) {
-  // write code here
+
+  for (const action of actions) {
+
+    switch (action.type) {
+      case 'addProperties':
+
+         Object.assign(state, action.extraData);
```

Comment:
this can also be achieved using [spread operator](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Spread_syntax) which is more common nowadays